initialize_package.sh

```diff
@@ -0,0 +1,367 @@
+#!/bin/bash
+set -euo pipefail
+IFS=$'\n\t'
+
+# Constants and help message.
+OLD_NAME="template_component_package"
+HELP_MESSAGE="Usage: $0 [--dry-run]
+This script initializes the package by prompting you to choose which component variants to include:
+  - Python Lifecycle Component
+  - Python Component
+  - C++ Lifecycle Component
+  - C++ Component
+
+Based on your choices, it will set up (or remove) the corresponding files and directories.
+
+Options:
+  -n, --dry-run    Display the actions without making any changes.
```

Comment:
Why is this the `-n` option? Feel like unless it's clear what it means, we just keep the verbose `--dry-run`